Production/plugins/L2EventTupleProducer.cc

```diff
@@ -0,0 +1,686 @@
+            ebHandle = ec_tmp;
+          }
+          else if ((ec_tmp->begin()->detid()).subdetId() == EcalEndcap) {
+            eeHandle = ec_tmp;
+          }
+        }
+        std::vector<edm::EDGetTokenT<EcalRecHitCollection> >::const_iterator i;
+        for (i = ecal_tokens.begin(); i != ecal_tokens.end(); i++) {
+          edm::Handle<EcalRecHitCollection> ec;
+          event.getByToken(*i, ec);
+        }
+        edm::Handle<HBHERecHitCollection> hbhe;
+        event.getByToken(hbhe_token, hbhe);
+
+        edm::Handle<HORecHitCollection> ho;
+        event.getByToken(ho_token, ho);
+
+        edm::Handle<HFRecHitCollection> hf;
+        event.getByToken(hf_token, hf);
+
+        edm::Handle<CaloTowerCollection> caloTowers;
+        event.getByToken(caloTowers_token, caloTowers);
+
+        edm::Handle<reco::TrackCollection> Tracks;
+        event.getByToken(Tracks_token, Tracks);
+
+        edm::Handle<reco::TrackCollection> pataTracks;
+        event.getByToken(pataTracks_token, pataTracks);
+
+        edm::Handle<reco::CaloJetCollection> caloTaus;
+        event.getByToken(caloTaus_token, caloTaus);
+        pat::JetCollection jets;
+
+        edm::ESHandle<CaloGeometry> Geometry = eventsetup.getHandle(Geometry_token);
+
+        caloRecHitCollections AllCaloRecHits;
+        AllCaloRecHits.hbhe= &*hbhe;
+        AllCaloRecHits.ho= &*ho;
+        AllCaloRecHits.hf= &*hf;
+        AllCaloRecHits.eb= &*ebHandle;
+        AllCaloRecHits.ee= &*eeHandle;
+        AllCaloRecHits.Geometry = &*Geometry;
+
+        edm::Handle<std::vector<reco::GenParticle>> hGenParticles;
+        if(isMC){
+            event.getByToken(genParticles_token, hGenParticles);
+            std::vector<reco_tau::gen_truth::GenLepton> GenLeptons = reco_tau::gen_truth::GenLepton::fromGenParticleCollection(*hGenParticles);
+            FillGenLepton(GenLeptons);
+          }
+
+        //check inputs
+        FillL1Objects(*l1Taus, l1_handle_map);
+        FillTracks(*Tracks, "track", *vertices);
+        FillTracks(*pataTracks, "patatrack", *pataVertices);
+        FillVertices(*vertices, "vert");
+        FillVertices(*pataVertices, "patavert");
+        FillCaloRecHit(AllCaloRecHits);
+        FillCaloTowers(*caloTowers);
+        FillCaloTaus(*caloTaus);
+        FillHLTResults(*Trigger_Results, module_names, trigger_names.triggerNames());
+        trainTuple.Fill();
+    }
+
+    virtual void endJob() override
+    {
+        L2EventTupleProducerData::ReleaseGlobalData();
+    }
+
+private:
+
+  void FillHLTResults(const edm::TriggerResults& Trigger_Results, const std::vector<std::string>& module_names, const std::vector<std::string>& trigger_names){
+    const int position_of_path  = std::distance(trigger_names.begin(), std::find( trigger_names.begin(), trigger_names.end(), defaultDiTauPath)) ;
+    const edm::HLTPathStatus& hlt_path_status = Trigger_Results.at(position_of_path);
+    trainTuple().defaultDiTauPath_result = hlt_path_status.state();
```

Comment:
- check that state is Pass or Fail, if something else -> throw an exception
- defaultDiTauPath_result should be `bool` and filled as `hlt_path_status.state() == edm::hlt::Pass`.